### PR TITLE
Add support for External Groups

### PR DESCRIPTION
**Pull request by kabesan**
Fixes: #2213 .
https://docs.github.com/en/enterprise-cloud@latest/rest/reference/teams#external-groups

### CODE REVIEW

File: github/teams.go [hunk 1]
```diff
@@ -854,3 +854,116 @@ func (s *TeamsService) CreateOrUpdateIDPGroupConnectionsBySlug(ctx context.Conte
...
+	externalGroup := new(ExternalGroup)
+	resp, err := s.client.Do(ctx, req, externalGroup)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return externalGroup, resp, nil
+}
+
+// RemoveConnectedExternalGroup removes the connection between on external group and a team
```
Comment (gmlewis) [inline github/teams.go:957]:
```suggestion
// RemoveConnectedExternalGroup removes the connection between an external group and a team.
```

File: github/teams_test.go [hunk 2]
```diff
@@ -1745,3 +1745,352 @@ func TestIDPGroup_Marshal(t *testing.T) {
...
+	if !cmp.Equal(externalGroup, want) {
+		t.Errorf("Teams.GetExternalGroup returned %+v, want %+v", externalGroup, want)
+	}
+
+	const methodName = "GetExternalGroup"
+	testBadOptions(t, methodName, func() (err error) {
+		_, _, err = client.Teams.GetExternalGroup(ctx, "", -1)
+		return err
+	})
+	testNewRequestAndDoFailure(t, methodName, client, func() (*Response, error) {
```
Comment (gmlewis) [inline github/teams_test.go:1830]: Please add a blank line before this test.

File: github/teams_test.go [hunk 3]
```diff
@@ -1745,3 +1745,352 @@ func TestIDPGroup_Marshal(t *testing.T) {
...
+	if !cmp.Equal(list, want) {
+		t.Errorf("Teams.ListExternalGroups returned %+v, want %+v", list, want)
+	}
+
+	const methodName = "ListExternalGroups"
+	testBadOptions(t, methodName, func() (err error) {
+		_, _, err = client.Teams.ListExternalGroups(ctx, "", nil)
+		return err
+	})
+	testNewRequestAndDoFailure(t, methodName, client, func() (*Response, error) {
```
Comment (gmlewis) [inline github/teams_test.go:1905]: Please add a blank line before this test.

File: github/teams_test.go [hunk 4]
```diff
@@ -1745,3 +1745,352 @@ func TestIDPGroup_Marshal(t *testing.T) {
...
+	if !cmp.Equal(externalGroup, want) {
+		t.Errorf("Teams.GetExternalGroup returned %+v, want %+v", externalGroup, want)
+	}
+
+	const methodName = "UpdateConnectedExternalGroup"
+	testBadOptions(t, methodName, func() (err error) {
+		_, _, err = client.Teams.UpdateConnectedExternalGroup(ctx, "", "", body)
+		return err
+	})
+	testNewRequestAndDoFailure(t, methodName, client, func() (*Response, error) {
```
Comment (gmlewis) [inline github/teams_test.go:2020]: Please add a blank line before this test.

File: github/teams_test.go [hunk 5]
```diff
@@ -1745,3 +1745,352 @@ func TestIDPGroup_Marshal(t *testing.T) {
...
+	if err != nil {
+		t.Errorf("Teams.RemoveConnectedExternalGroup returned error: %v", err)
+	}
+
+	const methodName = "RemoveConnectedExternalGroup"
+	testBadOptions(t, methodName, func() (err error) {
+		_, err = client.Teams.RemoveConnectedExternalGroup(ctx, "", "")
+		return err
+	})
+	testNewRequestAndDoFailure(t, methodName, client, func() (*Response, error) {
```
Comment (gmlewis) [inline github/teams_test.go:2074]: Please add a blank line before this test.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @kabesan !
Just a few tweaks, please, then we should be ready for a second LGTM before merging.

**Comment by gmlewis** [comment]
Also, please make sure to run unit tests before pushing (not force-pushing) your upcoming changes.
We squash and merge, so don't worry about the number of commits.
And finally, make sure to run `go generate ./...` and push those changes.
See our CONTRIBUTING.md file for more details.

**Comment by kabesan** [comment]
@gmlewis Thank your review.
I fiexed, Please re-review.

File: github/teams_test.go [hunk 6]
```diff
@@ -1745,3 +1746,359 @@ func TestIDPGroup_Marshal(t *testing.T) {
...
+		}`)
+	})
+
+	ctx := context.Background()
+	externalGroup, _, err := client.Teams.GetExternalGroup(ctx, "o", 123)
+	if err != nil {
+		t.Errorf("Teams.GetExternalGroup returned error: %v", err)
+	}
+
+	updatedAt, _ := time.Parse(time.RFC3339, "2021-01-24T11:31:04-06:00")
```
Comment (gmlewis) [inline github/teams_test.go:1793]: Here, and everywhere else you use `time.Parse`, please change this to use the already-existing `referenceTime` value created in: `github/timestamp_test.go`.

File: github/teams.go [hunk 7]
```diff
@@ -854,3 +854,116 @@ func (s *TeamsService) CreateOrUpdateIDPGroupConnectionsBySlug(ctx context.Conte
...
+type ExternalGroupTeam struct {
+	TeamID   *int64  `json:"team_id,omitempty"`
+	TeamName *string `json:"team_name,omitempty"`
+}
+
+// ExternalGroup represents an external group.
+type ExternalGroup struct {
+	GroupID   *int64                 `json:"group_id,omitempty"`
+	GroupName *string                `json:"group_name,omitempty"`
+	UpdatedAt *Timestamp                `json:"updated_at,omitempty"`
```
Comment (gmlewis) [inline github/teams.go:876]: Please run this file through `go fmt` to fix the indentation.

**Review by gmlewis — CHANGES_REQUESTED** [review]

**Review by gmlewis — APPROVED** [review]
Thank you, @kabesan !
LGTM.

Awaiting second LGTM before merging.

(Please note that ALL other contributors to this repo are welcome to provide the second PR review/comment/approval that we need for merging and that we are not waiting for any particular reviewer unless otherwise noted.)

**Comment by kabesan** [comment]
Fixed for diff coverage!

**Review by gmlewis — APPROVED** [review]

**Comment by kabesan** [comment]
@gmlewis Hi
Is second reviewer not active?
Can you add other reviewers?

**Comment by gmlewis** [comment]
> @gmlewis Hi Is second reviewer not active? Can you add other reviewers?

I am no longer able to add other reviewers, but any other contributor to this repo is welcome to review.

**Comment by kabesan** [comment]
Hi @pierce-m
Can you review this PR?
I'm waiting for second review.

**Comment by pierce-m** [comment]
@kabesan Yeah I'll take a look, but probably won't be able to until after the winter holidays

**Review by pierce-m — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @pierce-m !
Merging.

**Pull request merged**